### PR TITLE
WFLY-6997 Test cases for JBERET-231 and JBERET-117

### PR DESCRIPTION
**Pull request by spyrkob**
JIRA: https://issues.jboss.org/browse/WFLY-6997

### CODE REVIEW

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/batch/analyzer/transactiontimeout/TestWriter.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import javax.batch.api.chunk.AbstractItemWriter;
+import javax.inject.Named;
+import java.util.logging.Logger;
+
+@Named
+public class TestWriter extends AbstractItemWriter {
+    private static final Logger LOG = Logger.getLogger(TestWriter.class.getName());
+    public void writeItems(java.util.List<Object> items) throws Exception {
+        for(Object o : items){
+            LOG.info(o.toString());
```
Comment (jamezp) [inline testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/batch/analyzer/transactiontimeout/TestWriter.java]: I think we should avoid logging these or log them as debug. CI servers don't always have a lot of disk space and logging unneeded data will increase usage of the disk space.

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/batch/analyzer/transactiontimeout/TransactionTimeoutSetupStep.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        executeReload(managementClient);
+
+        awaitReload(managementClient);
+    }
+
+    @Override
+    public void tearDown(ManagementClient managementClient, String serverId) throws Exception {
+        setTimeout(managementClient, 300);
+
+        executeReload(managementClient);
```
Comment (jamezp) [inline testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/batch/analyzer/transactiontimeout/TransactionTimeoutSetupStep.java]: This method and the `awaitReload()` should be removed in favor of `org.jboss.as.test.shared.ServerReload.executeReloadAndWaitForCompletion()`

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/batch/analyzer/transactiontimeout/TestPartitionMapper.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+import javax.batch.api.BatchProperty;
+import javax.batch.api.partition.PartitionMapper;
+import javax.batch.api.partition.PartitionPlan;
+import javax.batch.api.partition.PartitionPlanImpl;
+import javax.inject.Inject;
+import javax.inject.Named;
+import java.util.Properties;
+
+@Named
+public class TestPartitionMapper implements PartitionMapper {
```
Comment (jamezp) [inline testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/batch/analyzer/transactiontimeout/TestPartitionMapper.java]: I don't see where this is being used. If it's not used it should probably be deleted.

**Comment by spyrkob** [comment]
The AsyncMethodTestCase failed due to https://issues.jboss.org/browse/WFLY-6021 - this should be fixed now

**Comment by jamezp** [comment]
This looks good to me.

**Pull request merged**